### PR TITLE
DB engine migration

### PR DESCRIPTION
**Pull request by aaannz**
- enhance smt-migration-upgrade to do DB engine migration
  at the end I decided to use the simplest way (alter table $t engine InnoDB, creating temporary
  tables proved to be more error prone
- add notification about needed DB migration during rpm upgrade
- add smt-migration-upgrade.service to be called after mysql.service and before apache2, all part
  of smt.target
- services are enabled during yast2 smt-server call of db/smt-db, but I don't know yet how to enable
  them after upgrade when yast2 smt-server is not run.
- some other fixes here and there

### CODE REVIEW

File: db/smt-db [hunk 1]
```diff
@@ -304,7 +305,7 @@ sub setup
     my ($login,$pass,$uid,$gid) = getpwnam("wwwrun");
     chown 0, $gid, "/etc/smt.conf";
 
-    `smt-sql --select-mode-direct /usr/share/schemas/smt/mysql/2.03/100-smt-tables.sql > /var/log/smt/smt-db-setup.log`;
+    `smt-sql --select-mode-direct /usr/share/schemas/smt/mysql/3.00/100-smt-tables.sql > /var/log/smt/smt-db-setup.log`;
```
Comment (mcalmer) [inline db/smt-db]: Hmm, the old is wrong but the new one is not even better :-)

I think we should create a "latest" directory".
Then we have only the "latest" and the migration directories "<oldversion>-<newversion>" .

A new setup happens always with the "latest" version and no migration is needed in this case.
Schema migrations are only for the running system.

@aaannz : What do you think?

Comment (aaannz) [inline db/smt-db]: Either latest, or sort it by version. At openQA we use DBIx which manages this by itself and it just sort it by schema version.
But I agree, this hard coded way is wrong.

**Comment by mcalmer** [comment]
smt-schema-upgrade changed a lot, but it looks like that it should work.
If not I think we will find out soon.

File: db/smt-db (hunk 1 above)
Comment (mcalmer) [inline db/smt-db]: smt11 used DBIx as well but it is horrible. I had massive trouble to write correct functions which works with this module.
The new way is a copy from SUSE Manager. SUSE Manager also has some kind of "latest" or "current" schema plus the upgrades.
I think nobody want to install with an old schema version which does not match the code.

Comment (aaannz) [inline db/smt-db]: Ok, how's the best way to do this? Create link in spec file? And I should probably document that somewhere too.

Comment (mcalmer) [inline db/smt-db]: - remove all directors with only a single number
- create a new directory named "latest" or "current" and put the 100-smt-tables there.
- for a new setup always call /usr/share/schemas/smt/mysql/latest/100-smt-tables.sql

If we change the schema in future, we create a migration directory <oldversion>-<new-version> and update 
latest/100-smt-tables.sql

This is my idea.

**Comment by aaannz** [comment]
added cefd3d9 with new upgrade layout. Building new packages for testing.

**Comment by mcalmer** [comment]
LGTM. give it a short try and then merge it.

**Comment by aaannz** [comment]
added 247438b Into migration_schema_version goes current version, right?

I tried some migrations and new installation (where the sql syntax error occurred fixed by 247438b). It is working well, there is one problem though and I don't know if it is by my setup or what.

When upgrading, it seems that schema-upgrade service is started after package is installed. During %post it correctly detect migration is needed and create update message but then db is migrated.
I confirmed manually that --check-version and --check-engine options does not trigger full migration.
Maybe as I packaged `smt.target` and have it enabled does this, but I thought systemd services are not automatically run during package installation.
Any thoughts how to prevent this? It takes 5m on my (low end) machine so it's not that bad bud still you are looking at stuck package installation.

**Comment by aaannz** [comment]
Ok, found it. `%service_del_postun` restarts services on upgrade. So I'll remove this for our services.

**Comment by mcalmer** [comment]
Good that you found it. I have not much clue about systemd :-)

**Comment by aaannz** [comment]
added b088441 where schema version is inserted after initial schema import. (+some other fixes)

**Pull request merged**